node/rest/src/routes.rs

```diff
@@ -247,6 +248,44 @@ impl<N: Network, C: ConsensusStorage<N>, R: Routing<N>> Rest<N, C, R> {
         Ok(ErasedJson::pretty(mapping_value))
     }
 
+    // GET /<network>/program/{programID}/mapping/{mappingName}?all={true}&metadata={true}
+    pub(crate) async fn get_mapping_values(
+        State(rest): State<Self>,
+        Path((id, name)): Path<(ProgramID<N>, Identifier<N>)>,
+        metadata: Query<Metadata>,
+    ) -> Result<ErasedJson, RestError> {
+        // If the `all` query parameter is set, return the full mapping.
+        if metadata.all.unwrap_or(false) {
+            // Retrieve the latest height.
+            let height = rest.ledger.latest_height();
```

Comment:
nit: the height can be looked up after the mapping values, just in case the next query fails

Reply:
also, since that one is only needed if `metadata.metadata` is `true`, it can happen directly in the applicable `ErasedJson`

Reply:
I brought it above because the `spawn_blocking` call consumes the `rest` object. 

So it was either move the height call up or to clone the `vm` or `finalize_store` objects (which are Arcs).

Reply:
Ah, makes sense; an `Arc` clone is probably a tiny bit faster, but it's unlikely to make any difference in this case.